### DISCUSSION TITLE
Spotify support ?

### DISCUSSION THREAD

**Discussion by eliascarella** [jeffvli/feishin#587] [Ideas]
First, let me start by saying thank you for making such a great piece of software ! 

Also, I know my suggestion might go against the whole idea of selhosting media but.... Would it be conceivable to have support for Spotify ? I'm a "hybrid" user and it would be awesome to browse my spotify albums/playlists as well as my self-hosted music. 
Maybe it's an option worth considering ?

Looking forward to your opinions ! 

Best,

**Comment by eliascarella** [comment]
Ping ?

**Comment by dev0T** [comment]
I don't think Spotify's API allows this kind of thing, I have been wanting to just migrate my playlists and it is pretty much impossible for example

**Reply by eliascarella** [reply]
Sorry didn't see your answer in time. 
It seems its now possible to play music with this new API: https://developer.spotify.com/documentation/web-playback-sdk
And retrieving Playlist (after auth by the user of course) seems possible as well: https://developer.spotify.com/documentation/web-api/reference/get-playlist

**Reply by kgarner7** [reply]
From a legal standpoint, if I understand the TOS correctly, it _would_ be legal (since each "service" is shown separately; some changes would need to be done to comply with some other display requirements).

However, from an implementation standpoint this is not going to happen (at least not from me):
1. Web SDK would considerably break the flow of how Feishin works (can't use with MPV, for example)
2. Feishin is meant for Navidrome/Jellyfin (maybe Subsonic [?]); other services are **definitely** out of scope
3. Personally, I have no intention of paying for premium (requirement to use third-party clients)
